docs/howto/composite-primary-key.txt

```diff
@@ -0,0 +1,130 @@
+=====================================
+How to define a composite primary key
+=====================================
+
+In Django, each model has a primary key.
+By default, this primary key consists of a single field.
+
+In most cases, a single-field primary key should suffice.
+In database design, however, defining a primary key that consists of multiple
+fields is sometimes necessary.
+
+To define a composite primary key, add the ``models.CompositePrimaryKey``
+field to your model:
+
+.. code-block:: pycon
+
+    class Tenant(models.Model):
+        pass
+
+    class User(models.Model):
+        pk = models.CompositePrimaryKey("tenant_id", "id")
+        tenant = models.ForeignKey(Tenant, on_delete=models.CASCADE)
+        id = models.IntegerField()
+
+This will instruct Django to create a composite ``PRIMARY KEY`` constraint
+instead of a single-field primary key.
+
+It's possible to define a composite primary key consisting of a single field.
+This is practically the same as setting ``primary_key=True`` on the field.
+
+.. code-block:: pycon
+
+    class Model(models.Model):
+        pk = models.CompositePrimaryKey("id")
+        id = models.IntegerField()
+
+Migrating to a composite primary key
+====================================
+
+As of now, Django doesn't offer built-in support for the migration of primary
+keys.
+
+This means once a table is created, its primary key cannot be changed via
+Django's migration operations, regardless of whether the table was created
+with a regular primary key or a composite primary key.
+
+If you would like to migrate an existing table from a single primary key to a
+composite primary key, follow your database backend's instructions to do so.
+
+Once the composite primary key is in place, add the ``CompositePrimaryKey``
+field to your model. This allows Django to recognize and handle the composite
+primary key appropriately.
+
+.. note::
+
+    While migration operations (e.g. ``AddField``, ``AlterField``) on
+    primary key fields are not supported, ``makemigrations`` will still detect
+    changes.
+
+    In order to avoid errors, it's recommended to apply such migrations with
+    ``--fake``.
```

Comment:
Maybe we should recommend [`SeparateDatabaseAndState`](https://docs.djangoproject.com/en/5.0/ref/migration-operations/#separatedatabaseandstate) here.

Reply:
Yes, I added a sentence.